### ISSUE TITLE
Long notes randomly get half deleted when editing on Android

### ISSUE THREAD

**Issue by Domarius** [laurent22/joplin#7861]
While editing on Android, at some random point, half the note will be gone. This automatically syncs and I have to go to Desktop and restore the previous version.

Also today it was scrolling around randomly in edit mode, adding blank new lines then scrolling up through the document all by itself. Then half the document was destroyed.  So this is probably related.

I've attached 2 screenshots using the "Outliner" plugin to give a quick visual overview on just how much content was lost.
Original note headings list (from restoring previous version)
![image](https://user-images.githubusercontent.com/7411541/222590653-dd5ef218-d16a-4a93-9d0a-112f9dd6ce21.png)
Corrupted note. Note where the content now stops.
![image](https://user-images.githubusercontent.com/7411541/222592379-c5db59b8-2b8c-4bc0-b826-198968b97d6f.png)

## Environment

Joplin version: Joplin 2.10.8, Database v42
Platform: Android
OS specifics: Samsung A22

## Steps to reproduce

1. Have a particularly long note I assume. Eg. the one that just got corrupted was 274 lines, 32,615 characters
2. Press edit
3. Start typing in the document for a few minutes.

## Logfile

![Screenshot_20230303_094354_Joplin](https://user-images.githubusercontent.com/7411541/222592066-2aae9a58-308b-43af-9520-79909898bc60.jpg)

**Comment by personalizedrefrigerator** [comment]
Are you using the Samsung keyboard with Grammarly integration enabled? If so, disable Grammarly and test again.

(Samsung Keyboard's Grammarly integration is known to cause issues like this in multi-line textfields. See the related [Dropbox issue](https://www.dropboxforum.com/t5/Delete-edit-and-organize/How-do-i-stop-Automatic-scrolling-to-bottom-of-page/td-p/589554)).

**Comment by Domarius** [comment]
I'm using the Samsung keyboard, yes. But no Grammarly is not enabled, I checked.

**Comment by personalizedrefrigerator** [comment]
Can confirm with Grammarly disabled. Disabling predictive text in Samsung Keyboard's settings seems to fix/work around the issue, but that certainly isn't ideal. I'm not experiencing this issue when using GBoard or Microsoft SwiftKey Keyboard.

I'm currently testing whether disabling spellcheck and/or upgrading the version of the CodeMirror dependency fixes the issue. **Edit**: Neither fixes the issue.

**Comment by personalizedrefrigerator** [comment]
[Related CodeMirror bug report](https://github.com/codemirror/dev/issues/1087).

**Comment by Domarius** [comment]
Thanks for working through this. Yes it seems disabling Predictive Text avoids the issue. Pretty easy to confirm since as soon as I re-enable it, the cursor jumps around all by itself, and I will predictably lose a bottom chunk of the document almost immediately while editing and adding new lines.  So when I disabled it and was able to type for almost a full minute with no issue, this tells me that is related to triggering the bug.

Joplin is great, I can really appreciate the amount of work gone into making it, and seeing this, I don't envy you having to deal with the various issues on various platforms!

**Comment by personalizedrefrigerator** [comment]
Quoting from the upstream issue:
> In the [ProseMirror thread](https://discuss.prosemirror.net/t/samsung-keyboard-within-android-webview-causes-a-spam-of-new-lines/5246/14):
>
>  >  Hey everyone – I spoke with Samsung and they said it’s a known issue with One UI. Fortunately, updating to One UI 5.1 appears to fix it. Let me know if updating solves it for you as well!

My tablet hasn't gotten this update yet (Samsung can be _very_ slow to roll out new updates to older devices). If you're able to update, let me know if it fixes the issue!

If not, here are some options:
1. Keep predictive text disabled
2. Use a different keyboard (e.g. GBoard)
3. Switch to the plain text editor in Joplin's settings (this may or may not help).

**Comment by Domarius** [comment]
Cool, will this just be a general update, or do I have to do something
special? I went into Updates and it said the software was up to date. I
have a Samsung A22.

I can destroy a note file instantly, every time, simply by going into edit
mode.

**Comment by Domarius** [comment]
This appears to be fixed, so far, and I'm not sure if it was a phone update or a Joplin update, it's been a while. I've been getting by for the past couple months with predictive text disabled, and I've only just enabled it and done a few tests. But the fact that it's working at all is very promising. The corruption used to be instant. It used to happen as soon as I opened the edit mode and placed the cursor inside the edit area, before even typing anything it would start freaking out.

I like the patch fix the Trix team came up with in that thread you linked [ProseMirror thread](https://discuss.prosemirror.net/t/samsung-keyboard-within-android-webview-causes-a-spam-of-new-lines/5246/14), that's crazy that such a solution was necessary.
